ledger-tool/src/main.rs

```diff
@@ -1448,13 +1619,116 @@ fn main() {
+                    if !bootstrap_validator_pubkeys.is_empty() {
+                        rehash = true;
+
+                        // Delete existing vote accounts
+                        for (address, mut account) in bank
+                            .get_program_accounts(Some(&solana_vote_program::id()))
+                            .into_iter()
+                        {
+                            println!("vote address: {} = {:?}", address, account);
+                            account.lamports = 0;
+                            bank.store_account(&address, &account);
+                        }
+
+                        // Add a new identity/vote/stake account for each of the provided bootstrap
+                        // validators
+                        let mut bootstrap_validator_pubkeys_iter =
+                            bootstrap_validator_pubkeys.iter();
+                        loop {
+                            let identity_pubkey = match bootstrap_validator_pubkeys_iter.next() {
+                                None => break,
+                                Some(identity_pubkey) => identity_pubkey,
+                            };
+                            let vote_pubkey = bootstrap_validator_pubkeys_iter.next().unwrap();
+                            let stake_pubkey = bootstrap_validator_pubkeys_iter.next().unwrap();
+
+                            bank.store_account(
+                                identity_pubkey,
+                                &Account::new(
+                                    bootstrap_validator_lamports,
+                                    0,
+                                    &system_program::id(),
+                                ),
+                            );
+
+                            let vote_account = vote_state::create_account_with_authorized(
+                                &identity_pubkey,
+                                &identity_pubkey,
+                                &identity_pubkey,
+                                100,
+                                VoteState::get_rent_exempt_reserve(&rent).max(1),
+                            );
+
+                            bank.store_account(
+                                stake_pubkey,
+                                &stake_state::create_account(
+                                    bootstrap_stake_authorized_pubkey
+                                        .as_ref()
+                                        .unwrap_or(&identity_pubkey),
+                                    &vote_pubkey,
+                                    &vote_account,
+                                    &rent,
+                                    bootstrap_validator_stake_lamports,
+                                ),
+                            );
+                            bank.store_account(vote_pubkey, &vote_account);
+                        }
+
+                        // Warp ahead at least two epochs to ensure that the leader schedule will be
```

Comment:
The forced warping is a little clunky here, but I favoured this over exposing more bank methods to tweak the leader schedule from the outside